cps2pc

```diff
@@ -0,0 +1,211 @@
+_supported_clang_features = _supported_gcc_features
+_supported_compile_features = {
+    'gcc': _supported_gcc_features, 'clang': _supported_clang_features}
+
+# Link features
+_supported_link_features = {x: {'threads': '-lpthread'}
+                            for x in _supported_kernels}
+
+#------------------------------------------------------------------------------
+def get_link_features(link_features, platform):
+
+    try:
+        features = []
+        for feature in link_features:
+            features.append(_supported_link_features[platform][feature])
+        return features
+    except KeyError as k:
+        raise Exception("Unknown link feature %s on %s" % (k.message, platform))
+
+#------------------------------------------------------------------------------
+def get_compile_features(compile_features, compiler):
+    features = []
+    # For loop with catch KeyError exception and check with
+    # WError, and W... flags
+    for feature in compile_features:
+        try:
+            features.append(_supported_compile_features[compiler][feature])
+        except KeyError as k:
+            ma = _re_we_features.match(feature)
+            if ma is not None:
+                features.append(
+                    _compilers_we_prefixes[compiler][ma.group(1)]+ma.group(5))
+                continue
+            raise Exception(
+                "Unknown link feature %s on %s" % (k.message, compiler))
+    return features
+
+#------------------------------------------------------------------------------
+def get_cpp_runtime_vendor(package):
+    if hasattr(package, 'platform') and hasattr(package.platform, 'runtime'):
+        try:
+            runtime = _supported_cpp_runtime_vendors[
+                package.platform.runtime.vendor]
+            return ['-l%s' % runtime]
+        except KeyError:
+            raise Exception("Unknown runtime %s" % runtime)
+    return []
+
+#------------------------------------------------------------------------------
+def get_definitions_flags(compile_definitions, compiler):
+    definitions = []
+    for definition in compile_definitions:
+        try:
+            ma = _re_definitions.match(definition)
+            # No need to test result `ma` as regex catches everything.
+            prefix = _compilers_definition_prefixes[compiler][ma.group(1)]
+            definitions.append(prefix + ma.group(2))
+        except:
+            raise Exception("Unknown definition argument %s" % definition)
+    return definitions
+
+#------------------------------------------------------------------------------
+def get_platform(package):
+    try:
+        return package.platform.kernel
+    except:
+        try:
+            output = subprocess.check_output(['uname', '-s'])
+            return output.strip().lower()
+        except:
+            return "unknown"
+
+#------------------------------------------------------------------------------
+def handle_kind(kind, location, link_location):
```

Comment:
FYI, `link_location` will be `None` iff `location` is `None`:

https://github.com/mwoehlke/pycps/blob/568ea872ff3196e9cdaa9b3f338f6565e875b985/cps.py#L97-L98

I'm not sure you even need to pass `location` to this. (CPS doesn't work this way, partly as an artifact of being "inspired" by CMake, but with PyCPS, you will only very rarely use `location` instead of `link_location`, and that almost exclusively on Windows.)

On a different note, "handle" is just about my least favorite word to use in function names...

Reply:
It seems reasonable to only pass `link_location` then. I simplified the code. I renamed the function with hopefully a better name.